### PR TITLE
Optimize favorites page fetch requests

### PR DESCRIPTION
**Pull request by Palakkgoyal**
This PR closes issue no. #607 
You can check the optimization by:
- remove all posts from favorites then, reload the favorite page, it will take no time for the optimized version to load but, very much for the unoptimized one.
- try the same thing by adding some posts to favorites, optimized version will take less time than the other one.

### CODE REVIEW

File: src/components/Favorite.jsx [hunk 1]
```diff
@@ -14,25 +15,38 @@ function Favorite() {
   const [posts, setPosts] = useState([]);
   const [loading, setLoading] = useState(true);
   const rowMode = useContext(RowModeContext);
+  const savedPostsArr = JSON.parse(localStorage.getItem("posts"));
```
Comment (narayan954) [inline src/components/Favorite.jsx:18]: this won't always exist, right? add a default value

**Review by narayan954 — CHANGES_REQUESTED** [review]
favourite page is giving error, can you try fixing that?
`Uncaught TypeError: Cannot destructure property 'caption' of 'post' as it is undefined`

**Comment by Palakkgoyal** [comment]
Okay. Going to fix these bugs.

**Comment by Palakkgoyal** [comment]
> favourite page is giving error, can you try fixing that? `Uncaught TypeError: Cannot destructure property 'caption' of 'post' as it is undefined`
Hey @narayan954 I fixed the console.log and also given a default value to the savedPostsArr but, I am not getting any error as you said related to caption.

**Comment by narayan954** [comment]
> > favourite page is giving error, can you try fixing that? `Uncaught TypeError: Cannot destructure property 'caption' of 'post' as it is undefined`

>  Hey @narayan954 I fixed the console.log and also given a default value to the savedPostsArr but, I am not getting any error as you said related to caption.

reviewing :)

**Comment by narayan954** [comment]
> > favourite page is giving error, can you try fixing that? `Uncaught TypeError: Cannot destructure property 'caption' of 'post' as it is undefined`
> 
> Hey @narayan954 I fixed the console.log and also given a default value to the savedPostsArr but, I am not getting any error as you said related to caption.

see :/
image

**Comment by narayan954** [comment]
> > > favourite page is giving error, can you try fixing that? `Uncaught TypeError: Cannot destructure property 'caption' of 'post' as it is undefined`
> > 
> > 
> > Hey @narayan954 I fixed the console.log and also given a default value to the savedPostsArr but, I am not getting any error as you said related to caption.
> 
> see :/ image

some posts are undefined, maybe that's because they're deleted?
image

**Review by narayan954 — APPROVED** [review]
looks good to me!

**Comment by Palakkgoyal** [comment]
Okay, got that its very necessary to check every aspect of code.

**Comment by narayan954** [comment]
> Okay, got that its very necessary to check every aspect of code.

yes! :)

**Pull request merged**